Source/CarthageKit/Errors.swift

```diff
@@ -133,6 +138,76 @@ public enum CarthageError {
 			return CarthageErrorCode.NoSharedSchemes.error([
 				NSLocalizedDescriptionKey: "Project \"\(project)\" has no shared schemes"
 			])
+
+		case let .DuplicateDependencies(duplicateDeps):
+			let deps = duplicateDeps
+				.sorted(<) // important to match expected order in test cases
+				.reduce("") { (acc, dep) in
+					"\(acc)\n\t\(dep)"
+				}
+			return CarthageErrorCode.DuplicateDependencies.error([
+				NSLocalizedDescriptionKey: "The following dependencies are duplicates:\(deps)"
+			])
+		}
+	}
+}
+
+/// A duplicate dependency, used in CarthageError.DuplicateDependencies.
+public struct DuplicateDependency {
+	/// The duplicate dependency as a project.
+	let project: ProjectIdentifier
+
+	/// The locations where the dependency was found as duplicate.
+	let locations: [String]
+
+	// The generated memberwise initialiser has internal access control and
+	// cannot be used in test cases, so we reimplement it as public. We are also
+	// sorting locations, which makes sure that we can match them in a
+	// test case.
+	public init(project: ProjectIdentifier, locations: [String]) {
+		self.project = project
+		self.locations = locations.sorted(<)
+	}
+}
+
+extension DuplicateDependency: Printable {
+	public var description: String {
+		return "\(project) \(printableLocations())"
+	}
+
+	private func printableLocations() -> String {
```

Comment:
I'd probably make this a computed property (`var`), but it doesn't matter much. If it's convenient to fix along the way, sure—if not, don't worry about it.